File: segundo/1er cuatri/PDOO/irrgarten_ruby/Irrgarten/shield.rb
# frozen_string_literal: true
require_relative 'dice'
require_relative 'combat_element'

class Shield < CombatElement
  def initialize(protection, uses)
    super(protection, uses)
  end

  def protect
    produce_effect
  end

  def to_s
    "S[#{super.to_s}]"
  end

  def inspect
    to_s
  end
end

File: segundo/1er cuatri/PDOO/Irrgarten_java/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class Shield extends CombatElement {
    public Shield(){
        super(1,1);
    }
    
    public Shield(float protection, int uses) {
        super(protection, uses);
    }
    
    public float protect() {
        return super.produceEffect();
    }
    
    @Override
    public String toString() {
        return "S[" + super.toString() + "]";
    }
}
